genesyscloud/outbound_ruleset/resource_genesyscloud_outbound_ruleset_utils.go

```diff
@@ -0,0 +1,255 @@
+package outbound_ruleset
+
+import (
+	gcloud "terraform-provider-genesyscloud/genesyscloud"
+	"terraform-provider-genesyscloud/genesyscloud/util/resourcedata"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v105/platformclientv2"
+)
+
+/*
+The resource_genesyscloud_outbound_ruleset_utils.go file contains various helper methods to marshal
+and unmarshal data into formats consumable by Terraform and/or Genesys Cloud.
+*/
+
+// getOutboundRulesetFromResourceData maps data from schema ResourceData object to a platformclientv2.Ruleset
+func getOutboundRulesetFromResourceData(d *schema.ResourceData) platformclientv2.Ruleset {
+	name := d.Get("name").(string)
+
+	return platformclientv2.Ruleset{
+		Name:        &name,
+		ContactList: gcloud.BuildSdkDomainEntityRef(d, "contact_list_id"),
+		Queue:       gcloud.BuildSdkDomainEntityRef(d, "queue_id"),
+		Rules:       buildDialerules(d.Get("rules").([]interface{})),
+	}
+}
+
+// buildDialerules maps a []interface{} into a Genesys Cloud *[]platformclientv2.Dialerrule
+func buildDialerules(rules []interface{}) *[]platformclientv2.Dialerrule {
+	rulesSlice := make([]platformclientv2.Dialerrule, 0)
+	for _, rule := range rules {
+		var sdkRule platformclientv2.Dialerrule
+		ruleMap := rule.(map[string]interface{})
```

Comment:
It's better to do a little `ok` check here. If the person using the resource defines an empty rule map like this `rule{}` they'll get a panic error here 
```golang
ruleMap, ok := rule.(map[string]interface{})
if !ok {
  continue
}
```

Reply:
just continuing on in the loop is grand